larpandoracontent/LArMonitoring/VisualParticleMonitoringAlgorithm.cc

```diff
@@ -0,0 +1,428 @@
+        {
+            LArPfoHelper::GetCaloHits(pPfo, view, caloHits);
+            LArPfoHelper::GetIsolatedCaloHits(pPfo, view, caloHits);
+        }
+        for (const CaloHit *pCaloHit : caloHits)
+        {
+            const HitType view{pCaloHit->GetHitType()};
+            if (view == HitType::TPC_VIEW_U)
+            {
+                if (isTrack)
+                    uTrackHits.emplace_back(pCaloHit);
+                else
+                    uShowerHits.emplace_back(pCaloHit);
+            }
+            else if (view == HitType::TPC_VIEW_V)
+            {
+                if (isTrack)
+                    vTrackHits.emplace_back(pCaloHit);
+                else
+                    vShowerHits.emplace_back(pCaloHit);
+            }
+            else
+            {
+                if (isTrack)
+                    wTrackHits.emplace_back(pCaloHit);
+                else
+                    wShowerHits.emplace_back(pCaloHit);
+            }
+        }
+        if (isTrack)
+        {
+            std::string suffix{std::to_string(pfoIdx) + "_T"};
+            if (!uTrackHits.empty())
+                PANDORA_MONITORING_API(VisualizeCaloHits(this->GetPandora(), &uTrackHits, "u_" + suffix, BLUE));
+            if (!vTrackHits.empty())
+                PANDORA_MONITORING_API(VisualizeCaloHits(this->GetPandora(), &vTrackHits, "v_" + suffix, BLUE));
+            if (!wTrackHits.empty())
+                PANDORA_MONITORING_API(VisualizeCaloHits(this->GetPandora(), &wTrackHits, "w_" + suffix, BLUE));
+        }
+        else
+        {
+            std::string suffix{std::to_string(pfoIdx) + "_S"};
+            if (!uShowerHits.empty())
+                PANDORA_MONITORING_API(VisualizeCaloHits(this->GetPandora(), &uShowerHits, "u_" + suffix, RED));
+            if (!vShowerHits.empty())
+                PANDORA_MONITORING_API(VisualizeCaloHits(this->GetPandora(), &vShowerHits, "v_" + suffix, RED));
+            if (!wShowerHits.empty())
+                PANDORA_MONITORING_API(VisualizeCaloHits(this->GetPandora(), &wShowerHits, "w_" + suffix, RED));
+        }
+        ++pfoIdx;
+    }
+
+    PANDORA_MONITORING_API(ViewEvent(this->GetPandora()));
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void VisualParticleMonitoringAlgorithm::MakeSelection(const MCParticleList *pMCList, const CaloHitList *pCaloHitList,
+    LArMCParticleHelper::MCContributionMap &mcMap)
+{
+    // Default reconstructability criteria are very liberal to allow for unfolded hierarchy
+    LArMCParticleHelper::PrimaryParameters parameters;
+    parameters.m_minPrimaryGoodHits = 2;
+    parameters.m_minHitsForGoodView = 1;
+    parameters.m_maxPhotonPropagation = std::numeric_limits<float>::max();
+    parameters.m_minHitSharingFraction = 0;
+    parameters.m_foldBackHierarchy = false;
```

Comment:
Are these OK to hardcode? Or worth making everything configurable, just as a matter of course? (Just to think about.)

Reply:
At some point I want to add configurability here, but I need to give some thought as to whether any of these parameters have implications for the way I prepare the visualisations, so I was thinking that would be for the next round of updates.